docs/en/upgradable-smart-contracts-openzeppelin-sdk.md

```diff
@@ -0,0 +1,220 @@
+* The `contracts` directory where we'll place our smart contracts.
+
+## Configuring LoomTruffleProvider
+
+Now, let's configure OpenZeppelin to use `loom-truffle-provider` and deploy to Loom Testnet (`extdev`). Replace the content of the `networks.js` file with the following snippet:
+
+```js
+const LoomTruffleProvider = require('loom-truffle-provider')
+const { sha256 } = require('js-sha256')
+const { CryptoUtils } = require('loom-js')
+function getLoomProviderWithMnemonic (seed, chainId, writeUrl, readUrl) {
+ const privateKeyUint8ArrayFromSeed = CryptoUtils.generatePrivateKeyFromSeed(new Uint8Array(sha256.array(seed)))
+ const privateKeyB64 = CryptoUtils.Uint8ArrayToB64(privateKeyUint8ArrayFromSeed)
+ return new LoomTruffleProvider(chainId, writeUrl, readUrl, privateKeyB64)
+}
+module.exports = {
+ networks: {
+ extdev: {
+ provider: function() {
+ const chainId = 'extdev-plasma-us1'
+ const writeUrl = 'http://extdev-plasma-us1.dappchains.com:80/rpc'
+ const readUrl = 'http://extdev-plasma-us1.dappchains.com:80/eth'
+ const seed = 'gravity top burden flip student usage spell purchase hundred improve check genre'
+ const loomTruffleProvider = getLoomProviderWithMnemonic(seed, chainId, writeUrl, readUrl)
+ // Quick implementation for web3_clientVersion
+ loomTruffleProvider.getProviderEngine().addCustomMethod('web3_clientVersion', () => {
+ return 'loom-js'
+ })
+ return loomTruffleProvider
+ },
+ networkId: '*',
+ },
+ },
+}
+```
+
+## Your First Contract
+
+> Note: The next sections are using almost the same steps as the OpenZeppelin tutorial on working with upgradable contracts. However, to make this tutorial easier to follow, we'll also present these steps here.
+
+In the `contracts` directory, let's create a file called `Counter.sol` with the following content:
+
+```solidity
+pragma solidity ^0.5.0;
+
+contract Counter {
+ uint256 public value;
+
+ function increase() public {
+ value++;
+ }
+}
+```
+
+Now let's deploy our smart contract and call the `increase` function by running the `yarn openzeppelin create` command and passing the following arguments:
+
+* The name of the smart contract (`Counter`)
+* the `--network` flag with the name of the network we want to deploy our smart contract to (`extdev`)
+* the `--init` flag with the name of the function we want to call once the contracts gets deployed (`increase`)
+
+```bash
+yarn openzeppelin create Counter --network extdev --init increase
+```
+
+
+```
+✓ Setting everything up to create contract instances
+✓ Instance created at 0x879d15B21DE431E9520D6880E46Df374F7abB2a3
+0x879d15B21DE431E9520D6880E46Df374F7abB2a3
+✨ Done in 360.73s.
+```
+
+> The `--init` flag is very useful because you can't use constructors with the OpenZeppelin SDK. For more details, please refer to the [OpenZeppelin SDK Upgrades Pattern]https://docs.openzeppelin.com/sdk/2.5/pattern#the-constructor-caveat) page.
+
+
+### Send-TX and Call
```

Comment:
`Send-TX and Call` -> `Calling Your Contract`

Reply:
Send-Tx and Call are both commands

Reply:
@enlight @eduardonunesp Yeah, both are commands. I made a few edits. Hope now it's clear.